### ISSUE TITLE
test_node_name failed on nightly_linux_debug

### ISSUE THREAD

**Issue by clalancette** [ros2/build_farmer#153]
Last night's [nightly_linux_debug](https://ci.ros2.org/view/nightly/job/nightly_linux_debug/1036/#showFailuresLink) had a single failure, which was test_node_name from test_rclcpp on connext.  Output:

```
(node_with_name) pid 16098: ['/home/rosbuild/ci_scripts/ws/build/test_rclcpp/node_with_name', 'rmw_connext_cpp'] (stderr > stdout, all > console)
(node_name_list) pid 16099: ['/home/rosbuild/ci_scripts/ws/build/test_rclcpp/node_name_list', 'node_with_name_rmw_connext_cpp'] (stderr > stdout, all > console)
[node_with_name] Starting node with name: node_with_name_rmw_connext_cpp
[node_name_list] Waiting for node with name: node_with_name_rmw_connext_cpp
[node_name_list] 
[node_name_list] - 
[node_name_list] - 
[node_name_list] - node_name_list
[node_name_list] - node_with_name_rmw_connext_cpp
[node_name_list]   found expected node name
(node_name_list) rc 0
() tear down
(node_with_name) signal SIGINT
[node_with_name] RTI Data Distribution Service Evaluation License issued to OSRF dthomas@osrfoundation.org For non-production use only.
[node_with_name] Expires on 05-nov-2019 See www.rti.com for more information.
[node_with_name] PRESPsService_removeMatchingRemoteEndpoints:!precondition: me == ((void *)0) || worker == ((void *)0) || remoteGuid == ((void *)0) || !( ((remoteGuid->prefix.hostId == (0x00000000)) && (remoteGuid->prefix.appId == (0x00000000)) && (remoteGuid->prefix.instanceId == (0x00000000)) && (remoteGuid->objectId != (0x00000000))) || ((remoteGuid->prefix.hostId != (0x00000000)) && (remoteGuid->prefix.appId != (0x00000000)) && (remoteGuid->prefix.instanceId != (0x00000000)) && (remoteGuid->objectId == (0x00000000))) )
[node_with_name] DISCParticipantDiscoveryPlugin_removeRemoteParticipantI:!remove remote endpoints of remote participant: 0X10F0002,0XE33E0000,0,0X1C1
[node_with_name] PRESInterParticipant_removeRemoteParticipant:[Local Participant: ac120002 3ee2 1] [Remote Participant: 1c1 10f0002 e33e0000 | Remote Endpoint: 0] Could not remove remote endpoint
[node_with_name] PRESInterParticipant_removeRemoteParticipant:[Local Participant: ac120002 3ee2 1] [Remote Participant: 1c1 10f0002 e33e0000 | Remote Endpoint: 0] Could not remove remote endpoint
[node_with_name] PRESParticipant_removeRemoteParticipantFromBuiltinChannels:Failed to remove remote participant 10f0002 e33e0000 0 to inter participant channel
[node_with_name] PRESParticipant_removeRemoteParticipant:!remove remote participant from builtin channels
[node_with_name] DISCParticipantDiscoveryPlugin_removeRemoteParticipantI:!remove remote participant: 0X10F0002,0XE33E0000,0,0X1C1
[node_with_name] DISCParticipantDiscoveryPlugin_removeRemoteParticipant:!remove remote participant: 0X10F0002,0XE33E0000,0,0X1C1
[node_with_name] DISCSimpleParticipantDiscoveryPluginReaderListener_onDataAvailable:!remove remote participant
[node_with_name] [INFO] [rclcpp]: signal_handler(signal_value=2)
[node_with_name] terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
[node_with_name]   what():  could not create service: rcl node's context is invalid, at /home/rosbuild/ci_scripts/ws/src/ros2/rcl/rcl/src/rcl/node.c:531
(node_with_name) rc -6
```

**Comment by MiguelCompany** [comment]
FYI, we found Connext could be too restrictive about part of the guidPrefix. We solved the error output problems on eProsima/Fast-RTPS#353, that has just been merged on master.

**Comment by sloretz** [comment]
The test fails when `rmw_connext_cpp` is used for `node_with_name` and `rmw_fastrtps_cpp` is used for `node_name_list`, but passes when that's reversed.

I'm guessing the long startup time with connext results in `node_with_name` advertised itself to fastrtps before it has finished initializing. In the original post it failed to create a service (parameters service?) while in another failure it failed to create a client (parameters client created by time_source?), so it appears to be receiving SIGINT at a random time. The signal handler for SIGINT causes the context to get shutdown, which results in an exception being thrown as the node tries to finish initializing itself.

<details><summary>Recent failure</summary>

```
test 23
    Start 23: test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp

23: Test command: /home/rosbuild/ci_scripts/venv/bin/python3 "-u" "/home/rosbuild/ci_scripts/ws/install/ament_cmake_test/share/ament_cmake_test/cmake/run_test.py" "/home/rosbuild/ci_scripts/ws/build/test_rclcpp/test_results/test_rclcpp/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp.xunit.xml" "--package-name" "test_rclcpp" "--output-file" "/home/rosbuild/ci_scripts/ws/build/test_rclcpp/ament_cmake_pytest/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp.txt" "--append-env" "LD_LIBRARY_PATH=/home/rosbuild/ci_scripts/ws/build/test_rclcpp" "--command" "/home/rosbuild/ci_scripts/venv/bin/python3" "-u" "-m" "pytest" "/home/rosbuild/ci_scripts/ws/build/test_rclcpp/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp_Debug.py" "-o" "cache_dir=/home/rosbuild/ci_scripts/ws/build/test_rclcpp/ament_cmake_pytest/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp/.cache" "--junit-xml=/home/rosbuild/ci_scripts/ws/build/test_rclcpp/test_results/test_rclcpp/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp.xunit.xml" "--junit-prefix=test_rclcpp"
23: Test timeout computed to be: 15
23: -- run_test.py: extra environment variables to append:
23:  - LD_LIBRARY_PATH+=/home/rosbuild/ci_scripts/ws/build/test_rclcpp
23: -- run_test.py: invoking following command in '/home/rosbuild/ci_scripts/ws/src/ros2/system_tests/test_rclcpp':
23:  - /home/rosbuild/ci_scripts/venv/bin/python3 -u -m pytest /home/rosbuild/ci_scripts/ws/build/test_rclcpp/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp_Debug.py -o cache_dir=/home/rosbuild/ci_scripts/ws/build/test_rclcpp/ament_cmake_pytest/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp/.cache --junit-xml=/home/rosbuild/ci_scripts/ws/build/test_rclcpp/test_results/test_rclcpp/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp.xunit.xml --junit-prefix=test_rclcpp
23: ============================= test session starts ==============================
23: platform linux -- Python 3.6.7, pytest-4.1.1, py-1.7.0, pluggy-0.8.1
23: cachedir: build/test_rclcpp/ament_cmake_pytest/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp/.cache
23: rootdir: /home/rosbuild/ci_scripts/ws, inifile:
23: plugins: rerunfailures-6.0, repeat-0.7.0, cov-2.6.1
23: collected 1 item
23: 
23: ../../../../build/test_rclcpp/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp_Debug.py F [100%]
23: 
23: =================================== FAILURES ===================================
23: ________________________________ test_node_name ________________________________
23: 
23:     def test_node_name():
23:         ld = LaunchDescriptor()
23:     
23:         cmd = ['/home/rosbuild/ci_scripts/ws/build/test_rclcpp/node_with_name']
23:         if 'rmw_connext_cpp':
23:           cmd += 'rmw_connext_cpp'.split(' ')
23:         env = None
23:         if 'rmw_connext_cpp':
23:             env = dict(os.environ)
23:             env['RMW_IMPLEMENTATION'] = 'rmw_connext_cpp'
23:         ld.add_process(
23:             cmd=cmd,
23:             name='node_with_name',
23:             env=env,
23:         )
23:     
23:         cmd = ['/home/rosbuild/ci_scripts/ws/build/test_rclcpp/node_name_list']
23:         if 'node_with_name_rmw_connext_cpp':
23:           cmd += 'node_with_name_rmw_connext_cpp'.split(' ')
23:         env = None
23:         if 'rmw_fastrtps_cpp':
23:             env = dict(os.environ)
23:             env['RMW_IMPLEMENTATION'] = 'rmw_fastrtps_cpp'
23:         ld.add_process(
23:             cmd=cmd,
23:             name='node_name_list',
23:             env=env,
23:             exit_handler=primary_exit_handler,
23:         )
23:     
23:         launcher = DefaultLauncher()
23:         launcher.add_launch_descriptor(ld)
23:         rc = launcher.launch()
23:     
23: >       assert rc == 0, "The launch file failed with exit code '" + str(rc) + "'. "
23: E       AssertionError: The launch file failed with exit code '-6'. 
23: E       assert -6 == 0
23: 
23: ../../../../build/test_rclcpp/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp_Debug.py:44: AssertionError
23: ----------------------------- Captured stdout call -----------------------------
23: (node_with_name) pid 14377: ['/home/rosbuild/ci_scripts/ws/build/test_rclcpp/node_with_name', 'rmw_connext_cpp'] (stderr > stdout, all > console)
23: (node_name_list) pid 14378: ['/home/rosbuild/ci_scripts/ws/build/test_rclcpp/node_name_list', 'node_with_name_rmw_connext_cpp'] (stderr > stdout, all > console)
23: [node_with_name] Starting node with name: node_with_name_rmw_connext_cpp
23: [node_name_list] Waiting for node with name: node_with_name_rmw_connext_cpp
23: [node_name_list] 
23: [node_name_list] - 
23: [node_name_list] - 
23: [node_name_list] - node_name_list
23: [node_name_list] - node_with_name_rmw_connext_cpp
23: [node_name_list]   found expected node name
23: (node_name_list) rc 0
23: () tear down
23: (node_with_name) signal SIGINT
23: [node_with_name] RTI Data Distribution Service Evaluation License issued to OSRF dthomas@osrfoundation.org For non-production use only.
23: [node_with_name] Expires on 05-nov-2019 See www.rti.com for more information.
23: [node_with_name] [INFO] [rclcpp]: signal_handler(signal_value=2)
23: [node_with_name] terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
23: [node_with_name]   what():  could not create client: rcl node's context is invalid, at /home/rosbuild/ci_scripts/ws/src/ros2/rcl/rcl/src/rcl/node.c:547
23: (node_with_name) rc -6
23:  generated xml file: /home/rosbuild/ci_scripts/ws/build/test_rclcpp/test_results/test_rclcpp/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp.xunit.xml 
23: =========================== 1 failed in 0.27 seconds ===========================
23: -- run_test.py: return code 1
23: -- run_test.py: verify result file '/home/rosbuild/ci_scripts/ws/build/test_rclcpp/test_results/test_rclcpp/test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp.xunit.xml'
1/1 Test #23: test_node_name__rmw_connext_cpp__rmw_fastrtps_cpp ...***Failed    0.72 sec
```
</details>

**Comment by sloretz** [comment]
fixed by ros2/system_tests#325
